cosmos/azure-spring-data-cosmos/cosmos-multi-tenant-samples/multi-tenant-by-container/src/main/java/com/azure/spring/data/cosmos/example/HomeController.java

```diff
@@ -0,0 +1,30 @@
+package com.azure.spring.data.cosmos.example;
+
+import com.azure.spring.data.cosmos.core.query.CosmosPageRequest;
+import com.azure.spring.data.cosmos.example.tenant.TenantStorage;
+import org.springframework.core.env.Environment;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HomeController {
+    private final TenantStorage tenantStorage;
+    private final UserRepository userRepository;
+    private final Environment env;
+
+    public HomeController(TenantStorage tenantStorage, UserRepository userRepository, Environment env) {
+        this.tenantStorage = tenantStorage;
+        this.userRepository = userRepository;
+        this.env = env;
+    }
+
+    @GetMapping(path = "/all")
+    public @ResponseBody String getAllTenantUsersAndOrders() {
+        //because order and user types are co-located with same partition key, we can hit the container once for both
```

Comment:
To make the sample as simple as possible, how about just using one repository instead of 2 repositorires (`UserRepository` and `OrderRepository`) in this sample? (I remember I mentioned this before.)

Reply:
Although the structure of the entities themselves in the sample might justify this (because they are contrived and simple), in typical cases of co-locating the entities would probably need to be separate as they are too different. It is already a paradigm shift for user to consider co-locating, so prefer to keep this separate for now (same as reason for not using profiles). If we do anything, it will be to make the entities more complex in the future. But again, prefer to iterate on this.